verticapy/core/vdataframe/_sys.py

```diff
@@ -416,56 +416,45 @@ def expected_store_usage(self, unit: str = "b") -> TableSample:
         ]
         for column in columns:
             ctype = self[column].ctype()
-            if (
-                (ctype[0:4] == "date")
-                or (ctype[0:4] == "time")
-                or (ctype[0:8] == "interval")
-                or (ctype == "smalldatetime")
-            ):
+            if ctype.startswith(("date", "time", "interval", "smalldatetime")):
                 maxsize, expsize = 8, 8
             elif "int" in ctype:
                 maxsize, expsize = 8, self[column].store_usage()
-            elif ctype[0:4] == "bool":
+            elif ctype.startswith("bool"):
                 maxsize, expsize = 1, 1
-            elif (
-                (ctype[0:5] == "float")
-                or (ctype[0:6] == "double")
-                or (ctype[0:4] == "real")
-            ):
+            elif ctype.startswith(("float", "double", "real")):
                 maxsize, expsize = 8, 8
-            elif (
-                (ctype[0:7] in ("numeric", "decimal"))
-                or (ctype[0:6] == "number")
-                or (ctype[0:5] == "money")
-            ):
+            elif ctype.startswith(("decimal", "number", "numeric", "money")):
                 try:
                     size = sum(
                         [
                             int(item)
                             for item in ctype.split("(")[1].split(")")[0].split(",")
                         ]
                     )
-                except:
+                except IndexError:
                     size = 38
                 maxsize, expsize = size, size
-            elif ctype[0:7] == "varchar":
+            elif ctype.startswith("varchar"):
                 try:
                     size = int(ctype.split("(")[1].split(")")[0])
-                except:
+                except IndexError:
                     size = 80
                 maxsize, expsize = size, self[column].store_usage()
-            elif (ctype[0:4] == "char") or (ctype[0:3] == "geo") or ("binary" in ctype):
+            elif ctype.startswith("geo") or ctype.endswith(
```

Comment:
You may want to double check this. Initially you were checking for [0:4] "char".
And initially there is no endswith as well.

Reply:
no it is what I want to check